index.html

```diff
@@ -2971,15 +2971,29 @@ <h3>Add Proof Set/Chain</h3>
 the <var>unsecuredDocument</var>.
           </li>
           <li>
+Let <var>matchingProofs</var> be an empty list.
+          </li>
+          <li>
 If <var>options</var> contains a <code>previousProof</code> attribute and that
-attribute is a string, check whether
-an element of the <var>allProofs</var> has a matching <code>id</code> attribute.
-If not, an error MUST be raised and SHOULD convey an error type of
-<a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>. If the
-<code>previousProof</code> attribute is an array, check that each element of that
-array is a string which matches the <code>id</code> attribute of an element
-of the <var>allProofs</var>. If not, an error MUST be raised and SHOULD convey
-an error type of <a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>.
+attribute is a string, add the element from <var>allProofs</var> with an
+<code>id</code> attribute matching <code>previousProof</code> to
+<var>matchingProofs</var>. If a proof with
+<code>id</code> equal to <code>previousProof</code>does not exist in
+<var>allProofs</var>, an error MUST be raised and SHOULD convey an error type of
+<a href="#PROOF_GENERATION_ERROR">PROOF_GENERATION_ERROR</a>.
+If the <code>previousProof</code> attribute is an array, add each element from
+<var>allProofs</var> with an <code>id</code> attribute that matches an element
+of that array. If any matches are missing, an error MUST be raised and SHOULD
```

Comment:
`If any matches are missing,` is unclear to me. Maybe something like this --
```suggestion
of that array. If any elements of <code>previousProof</code> array have an
<code>id</code> attribute that does not match the <code>id</code> attribute
of any element of <var>allProofs</var>, an error MUST be raised and SHOULD
```